azuki-script-generation/src/main/kotlin/com/anaplan/engineering/azuki/script/generation/ScriptGenerationCheck.kt

```diff
@@ -2,6 +2,38 @@ package com.anaplan.engineering.azuki.script.generation
 
 import com.anaplan.engineering.azuki.core.system.Check
 
-interface ScriptGenerationCheck : Check {
-    fun getCheckScript(): String
+interface ScriptGenerationCheck<E : ScriptGenerationEnvironment> : Check {
+
+    /**
+     * Composes the check into the environment.
+     * Returns a resolver which we can later use to get any composed checks for this check from the environment.
+     * The default skips composition entirely.
+     */
+    fun composeInto(environment: E): ComposedCheckResolver<E> = ComposedCheckResolver { listOf(this) }
```

Comment:
❓ I could make this a separate descendant interface and instead do a

```kotlin
if (it is ComposableScriptGenerationCheck) it.composeInto(environment)
else ComposedCheckResolver { listOf(it) }
```

which would make the intent a bit cleaner... but I'm not sure if changing inheritance into type assertions is a code smell.